npm/parseSelector_test.go

```diff
@@ -555,4 +559,389 @@ func TestFlattenNameSpaceSelector(t *testing.T) {
+						"a",
+					},
+				},
+				metav1.LabelSelectorRequirement{
+					Key:      "ns",
+					Operator: metav1.LabelSelectorOpIn,
+					Values: []string{
+						"t",
+					},
+				},
+			},
+			MatchLabels: map[string]string{
+				"c": "d",
+				"a": "b",
+			},
+		},
+		metav1.LabelSelector{
+			MatchExpressions: []metav1.LabelSelectorRequirement{
+				metav1.LabelSelectorRequirement{
+					Key:      "testExists",
+					Operator: metav1.LabelSelectorOpExists,
+					Values:   []string{},
+				},
+				metav1.LabelSelectorRequirement{
+					Key:      "testIn",
+					Operator: metav1.LabelSelectorOpIn,
+					Values: []string{
+						"frontend",
+					},
+				},
+				metav1.LabelSelectorRequirement{
+					Key:      "pod",
+					Operator: metav1.LabelSelectorOpIn,
+					Values: []string{
+						"a",
+					},
+				},
+				metav1.LabelSelectorRequirement{
+					Key:      "ns",
+					Operator: metav1.LabelSelectorOpIn,
+					Values: []string{
+						"y",
+					},
+				},
+			},
+			MatchLabels: map[string]string{
+				"c": "d",
+				"a": "b",
+			},
+		},
+		metav1.LabelSelector{
+			MatchExpressions: []metav1.LabelSelectorRequirement{
+				metav1.LabelSelectorRequirement{
+					Key:      "testExists",
+					Operator: metav1.LabelSelectorOpExists,
+					Values:   []string{},
+				},
+				metav1.LabelSelectorRequirement{
+					Key:      "testIn",
+					Operator: metav1.LabelSelectorOpIn,
+					Values: []string{
+						"frontend",
+					},
+				},
+				metav1.LabelSelectorRequirement{
+					Key:      "pod",
+					Operator: metav1.LabelSelectorOpIn,
+					Values: []string{
+						"b",
+					},
+				},
+				metav1.LabelSelectorRequirement{
+					Key:      "ns",
+					Operator: metav1.LabelSelectorOpIn,
+					Values: []string{
+						"t",
+					},
+				},
+			},
+			MatchLabels: map[string]string{
+				"c": "d",
+				"a": "b",
+			},
+		},
+		metav1.LabelSelector{
+			MatchExpressions: []metav1.LabelSelectorRequirement{
+				metav1.LabelSelectorRequirement{
+					Key:      "testExists",
+					Operator: metav1.LabelSelectorOpExists,
+					Values:   []string{},
+				},
+				metav1.LabelSelectorRequirement{
+					Key:      "testIn",
+					Operator: metav1.LabelSelectorOpIn,
+					Values: []string{
+						"frontend",
+					},
+				},
+				metav1.LabelSelectorRequirement{
+					Key:      "pod",
+					Operator: metav1.LabelSelectorOpIn,
+					Values: []string{
+						"b",
+					},
+				},
+				metav1.LabelSelectorRequirement{
+					Key:      "ns",
+					Operator: metav1.LabelSelectorOpIn,
+					Values: []string{
+						"y",
+					},
+				},
+			},
+			MatchLabels: map[string]string{
+				"c": "d",
+				"a": "b",
+			},
+		},
+	}
+
+	if !reflect.DeepEqual(expectedSelectors, testSelectors) {
+		t.Errorf("TestFlattenNameSpaceSelector failed @ 2nd selector deepEqual check.\n Expected: %+v \n Actual: %+v", expectedSelectors, testSelectors)
+	}
+}
+
+func TestFlattenNameSpaceSelectorWoMatchLabels(t *testing.T) {
+	firstSelector := &metav1.LabelSelector{
```

Comment:
Same here. You can make it concise and better readability by using variables to avoid unnecessary repeat.